sdks/java/core/src/main/java/org/apache/beam/sdk/io/Write.java

```diff
@@ -119,18 +118,9 @@ public void populateDisplayData(DisplayData.Builder builder) {
       builder
           .add(DisplayData.item("sink", sink.getClass()).withLabel("Write Sink"))
           .include("sink", sink)
-          .addIfNotDefault(
-              DisplayData.item("numShards", getNumShards()).withLabel("Fixed Number of Shards"),
-              0);
-    }
-
-    /**
-     * Returns the number of shards that will be produced in the output.
-     *
-     * @see Write for more information
-     */
-    public int getNumShards() {
-      return numShards;
+          .addIfNotNull(
+              DisplayData.item("sharding", getSharding() == null ? null : getSharding().getClass()))
+          .include("sharding", getSharding());
```

Comment:
Cleaned this up. include does indeed do that, but also doesn't permit null (and getSharding can of course be null)